tst/EnergyPlus/unit/OutputProcessor.unit.cc

```diff
@@ -3167,8 +3172,148 @@ namespace EnergyPlus {
 				"12,11,Electricity:Facility [J] !RunPeriod [Value,Min,Month,Day,Hour,Minute,Max,Month,Day,Hour,Minute]",
 			} ) );
 
+			TimeValue.deallocate(); 
 		}
 
+		TEST_F( EnergyPlusFixture, OutputProcessor_ResetAccumulationWhenWarmupComplete )
+		{
+			std::string const idf_objects = delimited_string( {
+				"Version,8.3;",
+				"Output:Variable,*,Zone Ideal Loads Supply Air Total Heating Energy,detailed;",
+				"Output:Meter:MeterFileOnly,DistrictHeating:HVAC,detailed;",
+				"Output:Variable,*,Zone Ideal Loads Supply Air Total Heating Energy,runperiod;",
+				"Output:Meter:MeterFileOnly,DistrictHeating:HVAC,r;",
+			} );
+
+			ASSERT_FALSE( process_idf( idf_objects ) );
+
+			GetReportVariableInput();
+			Array1D< ZonePurchasedAir > PurchAir; // Used to specify purchased air parameters
+			PurchAir.allocate( 1 );
+			SetupOutputVariable( "Zone Ideal Loads Supply Air Total Heating Energy [J]", PurchAir( 1 ).TotHeatEnergy, "System", "Sum", PurchAir( 1 ).Name, _, "DISTRICTHEATING", "Heating", _, "System" );
+
+			// Setup so that UpdateDataandReport can be called.
+			DataGlobals::DayOfSim = 365;
+			DataGlobals::DayOfSimChr = "365";
+			DataEnvironment::Month = 12;
+			DataEnvironment::DayOfMonth = 31;
+			DataEnvironment::DSTIndicator = 0;
+			DataEnvironment::DayOfWeek = 3;
+			DataEnvironment::HolidayIndex = 0;
+			DataGlobals::HourOfDay = 24;
+			DataGlobals::NumOfDayInEnvrn = 365;
+			DataGlobals::MinutesPerTimeStep = 10;
+
+			if ( DataGlobals::TimeStep == DataGlobals::NumOfTimeStepInHour ) {
+				DataGlobals::EndHourFlag = true;
+				if ( DataGlobals::HourOfDay == 24 ) {
+					DataGlobals::EndDayFlag = true;
+					if ( ( !DataGlobals::WarmupFlag ) && ( DataGlobals::DayOfSim == DataGlobals::NumOfDayInEnvrn ) ) {
+						DataGlobals::EndEnvrnFlag = true;
+					}
+				}
+			}
+
+			if ( DataEnvironment::DayOfMonth == WeatherManager::EndDayOfMonth( DataEnvironment::Month ) ) {
+				DataEnvironment::EndMonthFlag = true;
+			}
+			TimeValue.allocate( 2 );  // should this be commented out since belong to anonymous namespace in different file?
+			auto timeStep = 1.0 / 6;
+			SetupTimePointers( "Zone", timeStep );
+			SetupTimePointers( "HVAC", timeStep );
+
+			SetTimeValueMinutes( 10, 10 );
```

Comment:
@JasonGlazer Check [here](https://github.com/NREL/EnergyPlus/blob/abfe61e944e518a9ed0ee4f5258318ffdbe93530/tst/EnergyPlus/unit/OutputProcessor.unit.cc#L2696) for example using...
```c++
TimeValue( 1 ).CurMinute = 50;
TimeValue( 2 ).CurMinute = 50;
```